tests/fixed_point/test_fp_class.py

```diff
@@ -0,0 +1,215 @@
+"""Tests for the FixedPoint class methods"""
+import math
+import unittest
+
+import elfpy.errors.errors as errors
+from elfpy.math import FixedPoint
+
+
+class TestFixedPoint(unittest.TestCase):
+    r"""Unit tests to verify that the FixedPoint class methods are correct."""
+
+    INF = FixedPoint("inf")
+    NEG_INF = FixedPoint("-inf")
+    NAN = FixedPoint("nan")
+
+    def test_init(self):
+        r"""Test initialization for FixedPoint numbers"""
+        # str == float
+        assert FixedPoint("5") == FixedPoint("5.0")
+        assert FixedPoint("50_000.0") == FixedPoint("50000.0")
+        assert FixedPoint("50_000.0") == FixedPoint("50000.0")
+        assert FixedPoint("5_340_070.0") == FixedPoint("5340070.0")
+        assert FixedPoint("-5") == FixedPoint(-5.0)
+        assert FixedPoint("5.1") == FixedPoint(5.1)
+        assert FixedPoint("5.01") == FixedPoint(5.01)
+        assert FixedPoint("5.000001") == FixedPoint(5.000001)
+        assert FixedPoint("5.000001000") == FixedPoint(5.000001)
+        # int == float
+        assert FixedPoint(5) == FixedPoint(5e-18)
```

Comment:
Why would this pass?

Reply:
This is answered in the docs for FixedPoint (which I now realize are not rendering in the docs page 🤦). It's a long story, so I'll leave it for when we meet and explain FixedPoint.